### PR TITLE
Transition to standard smart pointers, part 13

### PR DESCRIPTION
**Pull request by SergioRAgostinho**
Partially addresses #2792.

All modules are compiling in my local environment. There were a couple of replacements which I was not able to perform. I battled a little with the compiler but eventually gave up.

### CODE REVIEW

File: gpu/kinfu_large_scale/tools/kinfuLS_app.cpp [hunk 1]
```diff
@@ -222,17 +223,17 @@ typename PointCloud<MergedT>::Ptr merge(const PointCloud<PointT>& points, const
...
-          return boost::shared_ptr<pcl::PolygonMesh>();
+          return pcl::PolygonMesh::Ptr();
 
   pcl::PointCloud<pcl::PointXYZ> cloud;
   cloud.width  = (int)triangles.size();
   cloud.height = 1;
   triangles.download(cloud.points);
 
-  boost::shared_ptr<pcl::PolygonMesh> mesh_ptr( new pcl::PolygonMesh() ); 
+  pcl::PolygonMesh::Ptr mesh_ptr( new pcl::PolygonMesh() );
```
Comment (taketwo) [inline gpu/kinfu_large_scale/tools/kinfuLS_app.cpp]: `PolygonMesh::Ptr mesh_ptr = pcl::make_shared<PolygonMesh> ();`

**Comment by SergioRAgostinho** [comment]
I had troubles performing the following replacement
https://github.com/PointCloudLibrary/pcl/blob/3e04ea7f5e8dae81db2411f74ba18dc7da448c3d/gpu/segmentation/include/pcl/gpu/segmentation/gpu_extract_labeled_clusters.h#L53-L58
It is causing an unintentional SFINAE and I can't figure out exactly how to overcome it. Give it a try if you can.

**Comment by taketwo** [comment]
I think the only solution here is to specify the point type explicitly at the calling site, i.e. `extractLabeledEuclideanClusters<pcl::PointXYZ>(pc, ...)`.

**Comment by SergioRAgostinho** [comment]
Thank you for the tip. The compiler messages pointed me completely in the wrong direction.

**Pull request merged**